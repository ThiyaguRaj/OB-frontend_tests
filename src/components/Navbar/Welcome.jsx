import React from 'react';
import { Link } from '@material-ui/core';
import { Alert } from '@material-ui/lab';
import './wel.css'

export default function Welcome(params) {
    return (
        <>
            <div className="con card card-body mt-0 bg-light">
                <div className="text-dark thd text-center mb-1">
                    <h1 className="head text-muted">
                        Lead to Revenue
                        Redefined
                </h1><hr />
                    <Alert severity="warning" className="text-center">An intelligent end-to-end
                    subscription management and
                    billing solution in ONE platform</Alert>
                </div>
            </div>
            <div className="con card card-body mt-2 bg-light">
                <div className="text-dark thd text-center mb-1">
                    <Alert className="text-center  bg-dark text-success">What we do ?</Alert>
                    <p className="text-left">
                        <small> In recent years, we have seen a significant shift where customers are increasingly wanting product experiences that are more optimized, instantaneous, flexible and streamlined.</small></p>
                </div>
            </div>
        </>
    )
}