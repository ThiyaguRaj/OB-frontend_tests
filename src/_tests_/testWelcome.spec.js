import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Alert } from '@material-ui/lab';
import { create } from 'react-test-renderer';
import Welcome from '../components/Navbar/Welcome';

describe("Test nav Component", () => {
    const root = shallow(<Welcome />);
    test("test Header", () => {
        expect(toJson(root)).toMatchSnapshot();
    })

    test("test header content", () => {
        const comp = create(<Welcome />).root;
        const ele = comp.findByType("h1");
        expect(ele.props.children).toEqual("Lead to Revenue Redefined")
    })

    test("Test divs", () => {
        const count = root.find("div");
        expect(count.length).toBe(4);
    })

    test("Test h1", () => {
        const count = root.find("h1");
        expect(count.length).toBe(1);
    })

    test("Test p", () => {
        const count = root.find("p")
        expect(count.length).toEqual(1)
    })

    test("Test small", () => {
        const count = root.find("small")
        expect(count.length).toEqual(1)
    })
})