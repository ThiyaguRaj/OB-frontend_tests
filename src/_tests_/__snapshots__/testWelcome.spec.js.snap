// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test nav Component test Header 1`] = `
<Fragment>
  <div
    className="con card card-body mt-0 bg-light"
  >
    <div
      className="text-dark thd text-center mb-1"
    >
      <h1
        className="head text-muted"
      >
        Lead to Revenue Redefined
      </h1>
      <hr />
      <WithStyles(ForwardRef(Alert))
        className="text-center"
        severity="warning"
      >
        An intelligent end-to-end subscription management and billing solution in ONE platform
      </WithStyles(ForwardRef(Alert))>
    </div>
  </div>
  <div
    className="con card card-body mt-2 bg-light"
  >
    <div
      className="text-dark thd text-center mb-1"
    >
      <WithStyles(ForwardRef(Alert))
        className="text-center  bg-dark text-success"
      >
        What we do ?
      </WithStyles(ForwardRef(Alert))>
      <p
        className="text-left"
      >
        <small>
           In recent years, we have seen a significant shift where customers are increasingly wanting product experiences that are more optimized, instantaneous, flexible and streamlined.
        </small>
      </p>
    </div>
  </div>
</Fragment>
`;
